docs: document how docker datasource resolve sourceUrl (#12468)
renovatebot/renovate: lib/datasource/docker/readme.md

> This datasource fetches the release note from a source repository specified according to the [pre-defined annotation keys of the OCI Image Format Specification](https://github.com/opencontainers/image-spec/blob/main/annotations.md).

Reviewer: Datasources do not fetch release notes, so this is incorrect. I think you mean that it identifies the source URL of a package?

Author: ```suggestion
This datasource allows to fetch the release note from a source repository specified according to the [pre-defined annotation keys of the OCI Image Format Specification](https://github.com/opencontainers/image-spec/blob/main/annotations.md).
```

@rarkins is it more accurate this way ? the next sentence explains how it works.

> This datasource fetches the release note from a source repository specified according to the [pre-defined annotation keys of the OCI Image Format Specification](https://github.com/opencontainers/image-spec/blob/main/annotations.md).

Reviewer: ```suggestion
This datasource identifies an image's source repository according to the [pre-defined annotation keys of the OCI Image Format Specification](https://github.com/opencontainers/image-spec/blob/main/annotations.md).
```

> This datasource will fetch a release note from a source repository specified according to the [pre-defined annotation keys](https://github.com/opencontainers/image-spec/blob/main/annotations.md) of the OCI Image Format Specification.

Reviewer 2: ```suggestion
This datasource fetches the release note from a source repository specified according to the [pre-defined annotation keys of the OCI Image Format Specification](https://github.com/opencontainers/image-spec/blob/main/annotations.md).
```

Use active voice.
Use better link text.

> Namely, it will extract the value of label `org.opencontainers.image.source` if it exist **on the latest stable tags**.

Reviewer 2: This sentence needs rewording, because right now it's unclear to me.

It will extract -> Who is doing the extracting? I think this refers to the Docker Datasource?

Author: ```suggestion
Namely, this datasource looks for the metadata of the **latest stable** image found on the docker registry and use the value of label `org.opencontainers.image.source` as sourceUrl.
```

Reviewer 2: ```suggestion
This datasource looks for the metadata of the **latest stable** image found on the Docker registry and uses the value of the label `org.opencontainers.image.source` as the `sourceUrl`.
```

How about this?

I've capitalize the proper noun `Docker` and used monospaced font for `sourceUrl`. I've also added some `the` words to fix the grammar.

> [Label Schema](https://label-schema.org/) is now superceded by OCI by annotations, and this datasource do not support `org.label-schema.vcs-url` label.

Reviewer 2: ```suggestion
The [Label Schema](https://label-schema.org/) is now superseded by OCI by annotations, and this datasource does not support the `org.label-schema.vcs-url` label.
```

Improve grammar, fix spelling for word.

> The [Label Schema](https://label-schema.org/) is now superseded by OCI annotations, and this datasource does not support the `org.label-schema.vcs-url` label.

Reviewer 2: ```suggestion
The [Label Schema](https://label-schema.org/) is superseded by OCI annotations, therefore this datasource does not support the `org.label-schema.vcs-url` label.
```

We can drop the word `now`.

I think the reason this datasource doesn't support `org.label-schema.vcs-url` is because upstream wants us to use OCI annotations? So I think we should use `therefore` to make relationship between the Label Schema being superseded as the reason that we don't support the Label Schema at all?

> The [Label Schema](https://label-schema.org/) is now superseded by OCI by annotations, and this datasource does not support the `org.label-schema.vcs-url` label.

Author: ```suggestion
The [Label Schema](https://label-schema.org/) is now superseded by OCI annotations, and this datasource does not support the `org.label-schema.vcs-url` label.
```

This datasource fetches the release note from a source repository specified according to the [pre-defined annotation keys of the OCI Image Format Specification](https://github.com/opencontainers/image-spec/blob/main/annotations.md).

Namely, it will extract the value of label `org.opencontainers.image.source` if it exist **on the latest stable tags**.

> The [Label Schema](https://label-schema.org/) is now superseded by OCI by annotations, and this datasource does not support the `org.label-schema.vcs-url` label.

Author: ```suggestion
The [Label Schema](https://label-schema.org/) is now superseded by OCI annotations, and this datasource does not support the `org.label-schema.vcs-url` label.
```
